src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseAsyncInAsync.cs

```diff
@@ -0,0 +1,265 @@
+
+        internal static DiagnosticDescriptor Descriptor = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableMessage,
+                                                                                      DiagnosticCategory.Reliability,
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        internal static DiagnosticDescriptor DescriptorNoAlternativeMethod = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                              s_localizableTitle,
+                                                                              s_localizableMessage,
+                                                                              DiagnosticCategory.Reliability,
+                                                                              RuleLevel.BuildWarning,
+                                                                              s_localizableDescription,
+                                                                              isPortedFxCopRule: false,
+                                                                              isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterOperationBlockStartAction(context =>
+            {
+                context.RegisterOperationAction(AnalyzeInvocation, OperationKind.Invocation);
+
+                context.RegisterOperationAction(AnalyzePropertyGetter, OperationKind.PropertyReference);
+            });
+        }
+
+        internal static void AnalyzePropertyGetter(OperationAnalysisContext context)
+        {
+            if (IsInTaskReturningMethodOrDelegate(context))
+            {
+                InspectMemberAccess(context, CommonInterests.SyncBlockingProperties);
+            }
+        }
+
+        internal static void AnalyzeInvocation(OperationAnalysisContext context)
+        {
+            if (IsInTaskReturningMethodOrDelegate(context))
+            {
+                if (context.Operation is not IInvocationOperation)
+                {
+                    return;
+                }
```

Comment:
This check might be cheaper than `IsInTaskReturningMethodOrDelegate`. If so, it should be checked first.  Perhaps `if (context.Operation is not IInvocationOperation && IsInTaskReturningMethodOrDelegate(context))`.